PWGDQ/Tasks/tableReader.cxx

```diff
@@ -351,6 +363,196 @@ struct AnalysisMuonSelection {
+
+  Filter filterBarrelTrackSelected = aod::dqanalysisflags::isBarrelSelectedDalitz > 0; // Dalitz candidates
+
+  OutputObj<THashList> fOutputList{"output"}; //! the histogram manager output list
+  OutputObj<TList> fStatsList{"Statistics"};  //! skimming statistics
+
+  std::map<int, uint8_t> dalitzmap;
+
+  std::vector<int> fTrackCutsId;
+  std::vector<AnalysisCompositeCut> fPairCuts;
+  int nCuts = 0;
+
+  HistogramManager* fHistMan;
+
+  void init(o2::framework::InitContext& context)
+  {
+    // Barrel track cuts
+    TString cutNamesStr = fConfigDalitzTrackCutsId.value;
+    if (!cutNamesStr.IsNull()) {
+      std::unique_ptr<TObjArray> objArray(cutNamesStr.Tokenize(","));
+      for (int icut = 0; icut < objArray->GetEntries(); ++icut) {
+        fTrackCutsId.push_back(std::stoi(objArray->At(icut)->GetName()));
+      }
+    }
+
+    // Pair cuts
+    TString cutNamesPairStr = fConfigDalitzPairCuts.value;
+    if (!cutNamesPairStr.IsNull()) {
+      std::unique_ptr<TObjArray> objArray(cutNamesPairStr.Tokenize(","));
+      for (int icut = 0; icut < objArray->GetEntries(); ++icut) {
+        fPairCuts.push_back(*dqcuts::GetCompositeCut(objArray->At(icut)->GetName()));
+      }
+    }
+
+    if (fTrackCutsId.size() != fPairCuts.size()) {
+      std::cout << "WARNING: YOU SHOULD PROVIDE THE SAME NUMBER OF TRACK AND PAIR CUTS" << std::endl;
+    }
+    nCuts = std::min(fTrackCutsId.size(), fPairCuts.size());
+
+    VarManager::SetUseVars(AnalysisCut::fgUsedVars); // provide the list of required variables so that VarManager knows what to fill
+    VarManager::SetDefaultVarNames();
+    fHistMan = new HistogramManager("analysisHistos", "aa", VarManager::kNVars);
+    fHistMan->SetUseDefaultVariableNames(kTRUE);
+    fHistMan->SetDefaultVarNames(VarManager::fgVariableNames, VarManager::fgVariableUnits);
+
+    // Create the histogram class names to be added to the histogram manager
+    TString histClasses = "";
+
+    if (fQA) {
+      for (int icut = 0; icut < nCuts; icut++) {
+        int trackCutId = fTrackCutsId.at(icut);
+        AnalysisCompositeCut pairCut = fPairCuts.at(icut);
+        histClasses += Form("TrackBarrel_track%d_%s;", trackCutId, pairCut.GetName());
+        histClasses += Form("PairsDalitz_track%d_%s;", trackCutId, pairCut.GetName());
+      }
+    }
+
+    DefineHistograms(fHistMan, histClasses, fConfigAddTrackPairHistogram);
+
+    fStatsList.setObject(new TList());
+    fStatsList->SetOwner(kTRUE);
+
+    // Dalitz selection statistics: one bin for each (track,pair) selection
+    TH1I* histTracks = new TH1I("TrackStats", "Dalitz selection statistics", nCuts, -0.5, nCuts - 0.5);
+    for (int icut = 0; icut < nCuts; icut++) {
+      int trackCutId = fTrackCutsId.at(icut);
+      AnalysisCompositeCut pairCut = fPairCuts.at(icut);
+      histTracks->GetXaxis()->SetBinLabel(icut + 1, Form("track%d_%s", trackCutId, pairCut.GetName()));
+    }
+    if (fQA) {
+      fStatsList->Add(histTracks);
+    }
```

Comment:
I would say this is not needed. You have already filled histograms for prefiltered tracks and pairs build in the prefiltering. is there any new information here beyond what I could extract by looking in the "TrackBarrel" classes above and check the number of entries? My suggestion is to remove this from the task.